doc/modules/language-guide/examples/CounterCertified.mo

```diff
@@ -0,0 +1,37 @@
+/// Simple counter (see `Counter.mo`), but uses `mo:base/CertifiedData` to
+/// implement the counter value as a certified variable.
+import CD "mo:base/CertifiedData";
+import Blob "mo:base/Blob";
+import Nat8 "mo:base/Nat8";
+import Nat32 "mo:base/Nat32";
+
+actor Counter {
+
+  var value : Nat32 = 0;
+
+  /// Helper; should be in base?
+  func blobOfNat32(n : Nat32) : Blob {
+    let byteMask : Nat32 = 0xff;
+    func byte(n : Nat32) : Nat8 { Nat8.fromNat(Nat32.toNat(n)) };
+    Blob.fromArray(
+      [byte(byteMask << 0 & value),
+       byte(byteMask << 8 & value),
+       byte(byteMask << 16 & value),
+       byte(byteMask << 24 & value)])
+  };
+
+  /// Update counter and certificate (via system).
+  public func inc() : async Nat32 {
+    value += 1;
+    CD.set(blobOfNat32(value));
+    return value;
+  };
+
+  /// Returns the current counter value,
+  /// and an unforgeable certificate (from the system) about its authenticity.
+  public query func get() : async ?{ value : Nat32; certificate : Blob } {
+    do ? {
+      { value; certificate = CD.getCertificate()! }
+    }
+  };
```

Comment:
I’d do explicit error handling here, to help readers understand when `getCertificate` returns `none`.

In fact, for best practice we want `get` to be callable via inter-canister call / update call, and then `getCertificate` fails, but it’s not really an error.

So maybe simply
```suggestion
  /// Returns the current counter value,
  /// and, if available, an unforgeable certificate (from the system) about its authenticity.
  /// When called via update call or inter-canister call, no certificate is present (and not needed,
  /// as in these cases the system already certifies the response)
  public query func get() : async { value : Nat32; certificate : ?Blob } {
    return { value; certificate = CD.getCertificate() }
  };
```

Sorry for not getting to show off `do ? {…}` :-)